front/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { AppState } from 'store';
+import { Redirect } from 'react-router-dom';
+
+export interface ProtectedRouteProps {
+  children: React.ReactNode;
+}
+
+const ProtectedRoute: React.FC<ProtectedRouteProps> = ({ children }: ProtectedRouteProps) => {
+  const isUserLoggedIn = useSelector((state: AppState) => state.user && state.user.isLoggedIn);
```

Comment:
I think we can pass the selector `selectIsLoggedIn` to useSelector here

Reply:
Hmnn... okay i can do that but still i will use it like **useSelector((state) => selectIsLoggedIn(state));**, i dont like using higher order functions is it okay by you??